lib/Reader/readerir.cpp

```diff
@@ -7799,12 +7944,45 @@ IRNode *GenIR::vectorCtorFromFloats(int VectorSize, IRNode *Vector,
   return 0;
 }
 
+IRNode *GenIR::vectorCtorFromArray(int VectorSize, IRNode *Vector,
+                                   IRNode *Array, IRNode *Index) {
+  llvm::LLVMContext &Context = *JitContext->LLVMContext;
+  Type *IntTy = Type::getInt32Ty(Context);
+  for (int Counter = 0; Counter < VectorSize; ++Counter) {
+    IRNode *ExtractedArg =
+        (IRNode *)LLVMBuilder->CreateLoad(genArrayElemAddress(
+            Array, Index, Vector->getType()->getVectorElementType()));
+    Vector = (IRNode *)LLVMBuilder->CreateInsertElement(Vector, ExtractedArg,
+                                                        Counter);
+    Index = (IRNode *)LLVMBuilder->CreateAdd(Index, ConstantInt::get(IntTy, 1));
+  }
+  return Vector;
+}
+
+IRNode *GenIR::vectorCtorFromPointer(int VectorSize, IRNode *Vector,
+                                     IRNode *Pointer, IRNode *Index) {
+  llvm::LLVMContext &Context = *JitContext->LLVMContext;
+  Type *IntTy = Type::getInt32Ty(Context);
+  Type *VectorElType = Vector->getType()->getVectorElementType();
+  Pointer = (IRNode *)LLVMBuilder->CreateBitCast(
```

Comment:
I think that this needs to be `CreatePointerCast`: IIRC `CreateBitCast` doesn't support casting pointers between different address spaces, so this will fail if `Pointer` is not in address space 0.